pydy/viz/scene.py

```diff
@@ -310,72 +271,98 @@ def generate_scene_dict(self, constant_map={}):
             frame.generate_numeric_transform_function(dynamic_variables,
                                                       constant_variables)
             frame.evaluate_transformation_matrix(dynamic_values,
-                                                         constant_values)
-
+                                                 constant_values)
 
             self._simulation_info.update(frame.generate_simulation_dict())
 
         for camera in self.cameras:
-            camera.generate_transformation_matrix(self._reference_frame,
-                                                 self._origin)
+            camera.generate_transformation_matrix(self.reference_frame,
+                                                  self.origin)
             camera.generate_numeric_transform_function(dynamic_variables,
-                                                      constant_variables)
+                                                       constant_variables)
             camera.evaluate_transformation_matrix(dynamic_values,
-                                                         constant_values)
+                                                  constant_values)
 
             self._simulation_info.update(camera.generate_simulation_dict())
 
         for light in self.lights:
-            light.generate_transformation_matrix(self._reference_frame,
-                                                 self._origin)
+            light.generate_transformation_matrix(self.reference_frame,
+                                                 self.origin)
             light.generate_numeric_transform_function(dynamic_variables,
                                                       constant_variables)
             light.evaluate_transformation_matrix(dynamic_values,
-                                                         constant_values)
+                                                 constant_values)
 
             self._simulation_info.update(light.generate_simulation_dict())
 
+        # TODO : This is bad practice. The method should either return the
+        # data or mutate the object, but not both.
         return self._simulation_info
 
     def generate_visualization_json_system(self, system, **kwargs):
-        """ Alternative method compatible with System class
-        for generating visualization JSON.
+        """Creates the visualization JSON files for the provided system.
 
         Parameters
-        ----------
-        system : object, pydy.System
+        ==========
+        system : pydy.system.System
+            A fully developed PyDy system that is prepared for the
+            ``.integrate()`` method.
+
+        Notes
+        =====
 
-        Keyword arguments are same as generate_visualization_json.
+        The optional keyword arguments are same as the
+        ``generate_visualization_json`` method.
 
         """
         if not isinstance(system, System):
-            self.system = None
```

Comment:
Good question. The tests are so sparse, I have no idea. But I think I thought about it when I removed it. Will look more closely.